apps/web/pages/api/book/event.ts

```diff
@@ -258,6 +258,15 @@ async function handler(req: NextApiRequest) {
         ...userSelect,
       })
     : eventType.users;
+  const isDynamicAllowed = !users.some((user) => {
+    return !user.allowDynamicBooking;
+  });
+  if (!isDynamicAllowed) {
+    throw new HttpError({
+      message: "Some of the users in this group do not allow dynamic booking",
+      statusCode: 401,
+    });
+  }
```

Comment:
Guards against dynamic booking a user (at `api/book/event`) who has the feature turned off

Reply:
I suggest `400 Bad Request` is a better status code for this situation; unauthorised may cause confusion.